maven-wrapper-distribution/src/resources/mvnw

```diff
@@ -194,7 +194,9 @@ else
       wrapperUrl="https://repo.maven.apache.org/maven2/org/apache/maven/wrapper/maven-wrapper/@@project.version@@/maven-wrapper-@@project.version@@.jar"
     fi
     while IFS="=" read -r key value; do
-      case "$key" in (wrapperUrl) wrapperUrl="$value"; break ;;
+      # Remove '\r' from value to allow usage on windows as IFS does not consider '\r' as a separator ( considers space, tab, new line ('\n'), and custom '=' )
+      safeValue=$(echo "$value" | sed 's/\r$//')
```

Comment:
using

```
`command`
```

may be favourable compared to

```
$(...)
```

...as some sources specify that $() doesn't appear to be supported by csh.

Reply:
-1. See: https://www.shellcheck.net/wiki/SC2006
We use `/bin/sh` which  should never point too `csh`, only  to posix compliant shells (bash, dash on Ubuntu).
If you do this, shellcheck will fail.

Reply:
Can you use `tr` instead of `sed` - it should be enough

Reply:
Or just use a bash variable expansion so we don't need to spawn another process.
Given everyone is on bash 4 now.

Reply:
> Or just use a bash variable expansion so we don't need to spawn another process. Given everyone is on bash 4 now.

That would be wrong assumption.
On macOS:
```
$ /bin/sh --version
GNU bash, version 3.2.57(1)-release (arm64-apple-darwin22)
Copyright (C) 2007 Free Software Foundation, Inc.
```

Reply:
> Or just use a bash variable expansion so we don't need to spawn another process. Given everyone is on bash 4 now.

NEVER expect something to be true in a target environment. That's Windows point of view.

Reply:
> Or just use a bash variable expansion

Didnt you say this was using /bin/sh? Surely that means there is no guarantee you have bash in the environment at all?

Reply:
correct, my original code had bash expansion but that fails posix.  Suggested change from 'sed' to 'tr' though makes sense.  Looking at that now.